### ISSUE TITLE
Vortex deleted my entire downloads folder, over 400 mods and 150+gigs

### ISSUE THREAD

**Issue by scrubborn** [sonic2kk/steamtinkerlaunch#611]
## System Information
- SteamTinkerLaunch version: 11.11
- Distribution: Garuda kde
- Installation Method: package manager (octopi)

## Issue Description
Basically i set up vortex with my fallout 4 with all my mods from my windows install, it was working fine. Next i decide to try getting my new vegas working and i originally chose mo2 for this. Mo2 was causing me issues daily so I swapped over to vortex. Vortex launched fine and i told it to setup new vegas, it spit out normal errors and nothing out of the ordinary, initial setup was fine. I already had downloaded mods for this though when using mo2, and i like storing my zip mods on my hdd instead of ssd, so i needed to swap the downloads directory.

I had already changed it for fallout 4 to point towards my hdd, where my mod collection was. Going to the download section with new vegas selected however it was pointing towards my boot drive which was obv wrong. But when I changed it, i went 1 folder too deep when assigning it(ie 'user/hdd/downloads/fnv' instead of 'user/hdd/downloads'). This had vortex try to move all my mods for all games including fallout 4, emphasis on try as it immediately started complaining about not having read/write perms. It wanted perms to user/.config/steamtinkerlaunch/vortex/downloads, and I had nothing open that was using that directory. This error gave me no option to grant perms to it, while ive had other perm errors in vortex that did have that option. Once the error closed all my mods were gone and the download directory had changed. The attached file is the screen I got immediately after the error closed, and when I realized my mods were gone. The 'old downloads folder' it mentions was empty except for a single txt file.

I don't need assistance here unless someone knows how to get my files out of oblivion, but god damn i did not think something like this was possible.

## Logs
My steamtinkerlaunch.log contains nothing from this encounter, infact it doesnt have anything that predates the last 14 minutes.

![Screenshot_Vortex_2](https://user-images.githubusercontent.com/38310465/193342476-b0390f1f-b283-4eac-8fa4-1739e21940a4.png)

**Comment by sonic2kk** [comment]
Hi there!

I'm not too familiar with Vortex but I'm just wondering two things:

1. Has the space on your hard drive that the mods would've taken up now been freed up? In other words, do you have 150GB now free on your hard drive? Or perhaps 150GB now taken up on another hard drive.
2. The permissions error should not happen as far as I'm aware, I don't know why Vortex wouldn't have permission to write to the folder you mentioned (that seems like a STL config folder which Vortex's exe gets downloaded to, not anything special). However, this could be a general issue with wine/vortex itself (even irrespective of wine). There may not be much we can do here.

I'm not too familiar with Vortex myself but your mods shouldn't have disappeared.  The error message says your downloads folder was copied, but that it didn't have permission to "clean up" (remove?) another folder.

If your drives show no files missing, there is likely _some_ way to recover them :) You might find a Vortex log somewhere in `~/.config/SteamTinkerLaunch/logs/Vortex` or a similarly named folder. I'm not at my PC atm so I can't verify but if there is a file in there that looks like it was modified at a reasonable time (e.g., 1 day ago) you could attach it. Though I'm not sure a log like that exists and contains much useful, I haven't fiddled around with Vortex in quite a while.

I would have a look around if you haven't already to see if you can find your mod files, but if you're missing 150GB on your drive there might not be much we can do. Hopefully this is not the case though!

**Comment by scrubborn** [comment]
1. I believe so, but I can't be positive as I was caught off guard and 150gb on a 2tb drive isn't that obvious. I'm not sure where it wouldve sent the files, it may have been attempting (wrongly) to move them to the .config folder i mentioned but that drive did not have enough space to store it.  

2. The perms errors continue today it seems. It is not saving any change to the downloads folder directory, but it still has my mod staging folder locked down correctly. I get the same error as earlier when I try to swap the downloads folder, and upon reboot of the app the change is lost. I got a pic of the error this time, and it was followed by the same failedsuccess screen that I already posted.

I can't find any vortex specific log in there but I found a log for fnv that mentions vortex in it. By the way, my HDD im trying to get this working with is ntfs. Dunno if it matters, and it was working perfectly earlier. 

While I havent found any of the missing files vortex did not nuke my mod staging folder, so I ran a command to zip all those files up individually to replace my downloads folder. Some mods are lost as they werent actively installed, but I have the majority.

Also I didnt mean to click the close button for this thread, whoops

![Screenshot_22](https://user-images.githubusercontent.com/38310465/193418894-37d72128-d0e4-4b3f-9b16-073f5b7139fc.png)
[Fallout New Vegas.log](https://github.com/frostworx/steamtinkerlaunch/files/9690756/Fallout.New.Vegas.log)

**Comment by scrubborn** [comment]
Ya know, I think this is atleast partly an ntfs issue. I am new to linux and coming from w10 so a ton of my data are on ntfs drives. I mounted them following this guide (https://github.com/ValveSoftware/Proton/wiki/Using-a-NTFS-disk-with-Linux-and-Windows) and things have seemed fine mostly till now, my games stored on ntfs launch via proton fine. I do however have a smaller drive that I setup for ext4 and I just tried moving the downloads folder to it, it did not give me an error and it appeared to work. I even put a mod in the folder and it picked it up just fine. However on reboot of vortex this change is lost and the directory is yet again "Z:\home\nau\.config\steamtinkerlaunch\vortex\downloads".

**Comment by sonic2kk** [comment]
First off,

> While I havent found any of the missing files vortex did not nuke my mod staging folder, so I ran a command to zip all those files up individually to replace my downloads folder. Some mods are lost as they werent actively installed, but I have the majority.

Whew, I'm glad. I understand the pain of losing a bunch of mods so even I'm a little relieved to hear this.

> The perms errors continue today it seems [...]

As you've kind of realised yourself (I was even in the middle of linking to the Proton wiki!) NTFS is **not** supported by Wine. Even if it *was* working you should not expect NTFS to work on Linux with Wine applications long-term. It is likely that Wine is having some difficulty reading/writing permissions to the NTFS drive - I have a 2TB NTFS HDD on my PC and every so often I have to reset the permissions when the drive is mounted (though it's gotten better with the new NTFS kernel driver). It's a real pain so if I even generally on Linux I get permission errors, it's very possible you might have permission errors when trying to use this folder with Wine.

I understand that having a bunch of data drives in NTFS is going to be awkward to convert to ext4 or another Linux-supported format (such as btrfs or zfs). That's just how it is unfortunately, NTFS support is improving massively especially with the kernel `ntfs3` driver, but it can have issues with Wine as you've seen even documented on the Proton wiki.

Out of curiosity, where is the mod staging folder?

As for the "drive in use" error, it is possible that this folder *is* actually in use - By SteamTinkerLaunch. I am not positive on this though but it could be that generally the `~/.config/steamtinkerlaunch` folder is in use by STL, including all the folders therein. However the question of why Vortex wants to write to that folder is a mystery to me. Perhaps it tries to default there for some reason since it can't write to your NTFS drive, though I'm not sure because...

> I do however have a smaller drive that I setup for ext4 and I just tried moving the downloads folder to it, it did not give me an error and it appeared to work. I even put a mod in the folder and it picked it up just fine. However on reboot of vortex this change is lost and the directory is yet again "Z:\home\nau.config\steamtinkerlaunch\vortex\downloads".

How do you launch Vortex with STL? I am not entirely sure how it works because I am not too familiar with it (on Linux or otherwise), but if it works like MO2 does with STL there'll be a button on the main menu and a requester dialog when you press the "Play" button. If you're using the main menu button, try pressing STL's "Play" button and seeing if that launches Vortex. Perhaps this issue is similar to a couple of prior issues with MO2 where using the Main Menu button may cause some issues - Though for Vortex I have no idea. Really it's just to see if Vortex tries to use a different folder if you press the "Play" button (if you aren't currently doing so).

<hr>

In short, NTFS is potentially causing the permissions issues and you should avoid using NTFS on Linux. But for why Vortex is trying to use the wrong folder, I'm not entirely sure. If we can get that solved, the solution to the overall problem will be to use ext4 (or another Linux filesystem, but ext4 is the most well supported). That comes after though, first we need to get Vortex to see the correct folder that you selected :smiley:

**Comment by scrubborn** [comment]
>Whew, I'm glad. I understand the pain of losing a bunch of mods so even I'm a little relieved to hear this.  
  
Thanks, yeah the thought of losing absolutely everything is heart-wrenching for me. After this though I'm surely taking more precautions to protect my mods.

>Out of curiosity, where is the mod staging folder?  
  
"Z:\home\nau\Games\Vortex\Staging\FO4" is how vortex sees it. Its on my boot drive.  
  
>How do you launch Vortex with STL?  
  
I have the prompt setup to ask and wait 5 seconds for a response if i want to boot into vortex.  
  
>As for the "drive in use" error, it is possible that this folder is actually in use - By SteamTinkerLaunch.   
  
I was thinking that as well, but I'm not sure I'm entirely sure. The fact that pointing it to my ext4 drive was a "success" makes me think its failing to move that single text file from the .config folder over to the ntfs disk and the error message is misreporting/isn't equipped to report this.  
  

So I've gotten it working I think. I still have no clue why it wont save changes to the downloads folder location, but I symlinked "Z:\home\nau\.config\steamtinkerlaunch\vortex\downloads" with my folder on my HDD that I wanted to store my mods on. The HDD is still ntfs, but it appears to be working and has survived me rebooting both the app and my pc just to be sure.  
  
If that sounds especially risky/stupid let me know, but this may be how I use it for the time being. I'm going to keep poking at it and see if it breaks.

**Comment by scrubborn** [comment]
Ok, ive run into something else. Vortex will not switch profiles which is an issue as when I try to mod fnv its defaulting to fo4, trying to swap over just has me stuck on "switching to profile: default". I see nothing happening but my cpu is hanging around 50-60% usage, that feels high for my r7 3800x with so little going on. I also just had this profile swapping peg me at 90% cpu, after restarting my pc though it hasnt gone that high again.  
  
edit- i already take it back, it finally swapped and i think it was just vortex being painfully slow. This action on windows takes under a minute but this took maybe 3+ minutes. I am still curious though what performance other people are getting out of vortex here and how its stressing their pcs.

**Comment by sonic2kk** [comment]
> The HDD is still ntfs, but it appears to be working and has survived me rebooting both the app and my pc just to be sure. If that sounds especially risky/stupid let me know, but this may be how I use it for the time being. I'm going to keep poking at it and see if it breaks.

I think "stupid" is too harsh :-) But risky, I would say so. But I am all too aware that it isn't really feasible to reformat such a big drive with a bunch of data on it. So to try and mitigate any risks, I would just make sure that you're using the newer `ntfs3` kernel driver, as opposed to the older NTFS-3G driver. I'm not sure if you manually set your `/etc/fstab` up or if your DE / disk manager manages that for you, so I'm not sure what driver you might be using. The tl;dr is that Linux Kernel 5.15 and above have a built in ntfs driver called `ntfs3`. Previously it used NTFS-3G which was ... a thing :-)

If you're comfortable doing so, run `uname -r` in a terminal and if this returns anything with `5.15` or higher: you can can check the `/etc/fstab` file (this is a system file so you might get warnings that you can't write to it - that's fine, we just want to view it) and see if any of the entries are using `ntfs3`. If they use `ntfs`/`ntfs-3g` (I can't remember exactly how it was called, sorry!) then you'll want to change it to use `ntfs3`.

This isn't a cure-all and you could already be using `ntfs3`, but if you aren't using it this could help with any potential problems that might come in future - It could also help with speed. I'm honestly not sure if distros now have defaulted to `ntfs3` or anything, so this is not to be taken as a guaranteed solution but more of something that could help alleviate *some* potential future problems.

And of course, taking care to back up your mods is always an excellent idea - NTFS on Linux or otherwise!

> This action on windows takes under a minute but this took maybe 3+ minutes. I am still curious though what performance other people are getting out of vortex here and how its stressing their pcs.

Vortex may have some issues on Linux. Remember that Vortex itself is currently unsupported on Linux and is using dotnet6 nowadays which has been a mixed bag. SteamTinkerLaunch tries to bridge the gap and there is still some ongoing work, but this should improve with time. I agree it shouldn't have this high CPU usage but likely nothing can be done about this on the Linux side. It could also be because you're using an NTFS drive, historically NTFS was always slower on Linux but `ntfs3` was meant to have alleviated that. Perhaps a mix of Wine not liking NTFS + the driver not being perfect in terms of speed is causing the problem, or contributing to it at least.

Good to know it can just take time though :) Thank you for updating!

**Comment by frostworx** [comment]
Thanks for the support @sonic2kk!
Apparently this is no steamtinkerlaunch issue _(therefore I already removed the "bug" label yesterday)_
Wrong user configuration often can lead to data loss. Thanks for sharing the story, but I see no reason to keep this "issue" open.
